Model/Plugins/MethodList.php

```diff
@@ -0,0 +1,210 @@
+<?php
+
+/**
+ * PAYONE Magento 2 Connector is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PAYONE Magento 2 Connector is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with PAYONE Magento 2 Connector. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * PHP version 5
+ *
+ * @category  Payone
+ * @package   Payone_Magento2_Plugin
+ * @author    FATCHIP GmbH <support@fatchip.de>
+ * @copyright 2003 - 2016 Payone GmbH
+ * @license   <http://www.gnu.org/licenses/> GNU Lesser General Public License
+ * @link      http://www.payone.de
+ */
+
+namespace Payone\Core\Model\Plugins;
+
+use Magento\Payment\Model\MethodList as OrigMethodList;
+use Magento\Payment\Model\MethodInterface;
+use Magento\Quote\Api\Data\AddressInterface;
+
+/**
+ * Plugin for Magentos MethodList class
+ */
+class MethodList
+{
+    /**
+     * PAYONE consumerscore request model
+     *
+     * @var \Payone\Core\Model\Api\Request\Consumerscore
+     */
+    protected $consumerscore;
+
+    /**
+     * Consumerscore helper
+     *
+     * @var \Payone\Core\Helper\Consumerscore
+     */
+    protected $consumerscoreHelper;
+
+    /**
+     * Checkout session
+     *
+     * @var \Magento\Checkout\Model\Session
+     */
+    protected $checkoutSession;
+
+    /**
+     * Constructor
+     *
+     * @param \Payone\Core\Model\Api\Request\Consumerscore $consumerscore
+     * @param \Payone\Core\Helper\Consumerscore            $consumerscoreHelper
+     * @param \Magento\Checkout\Model\Session              $checkoutSession
+     */
+    public function __construct(
+        \Payone\Core\Model\Api\Request\Consumerscore $consumerscore,
+        \Payone\Core\Helper\Consumerscore $consumerscoreHelper,
+        \Magento\Checkout\Model\Session $checkoutSession
+    ) {
+        $this->consumerscore = $consumerscore;
+        $this->consumerscoreHelper = $consumerscoreHelper;
+        $this->checkoutSession = $checkoutSession;
+    }
+
+    /**
+     * Filter methods by the worst score
+     *
+     * @param  MethodInterface[] $aPaymentMethods
+     * @param  string            $sWorstScore
+     * @return MethodInterface[]
+     */
+    protected function filterMethodsByScore($aPaymentMethods, $sWorstScore)
```

Comment:
This method is too complex, please refactor: https://scrutinizer-ci.com/g/PAYONE-GmbH/magento-2/inspections/f57e816f-b8e5-4e29-b0a2-aa3d5399368d/code-structure/operation/Payone%5CCore%5CModel%5CPlugins%5CMethodList%3A%3AfilterMethodsByScore